challenge-manager/challenge-tree/paths.go

```diff
@@ -110,22 +109,25 @@ func (ht *RoyalChallengeTree) IsConfirmableEssentialNode(
 	}
 	// An essential node is confirmable if all of its essential paths
 	// down the tree have a path weight >= the confirmation threshold.
-	// To do this, we compute the path weight of each path and insert
-	// into a min heap. Then, it is sufficient to check that the minimum
-	// element of the heap is >= the confirmation threshold.
-	pathWeights := newPathWeightMinHeap()
+	// To do this, we compute the path weight of each path and find the minimum.
+	// Then, it is sufficient to check that the minimum is >= the confirmation threshold.
+	minWeight := uint64(math.MaxUint64)
 	for _, timers := range essentialTimers {
 		pathWeight := uint64(0)
 		for _, timer := range timers {
-			pathWeight += timer
+			if timer == math.MaxUint64 {
+				pathWeight = math.MaxUint64
+			} else {
+				if pathWeight != math.MaxUint64 {
+					pathWeight += timer
```

Comment:
I think this can still overflow MaxUint64.

Imagine:
```
pathWeight := math.MaxUint64 - uint64(2)
timer := uint64(7)

timer == math.MaxUint64.     // false: puts us in the else branch on line 121
pathWeight != math.MaxUint64 // true:  puts us on line 122
pathWeight += timer          // pathWeight == 4, but we want pathWeight == math.MaxUint64
```

You could pull over `arbmath.SaturatingUAdd` from the nitro repo and simplify this to:
```
pathWeight = arbmath.SaturatingUAdd(pathWeight, timer)
```